src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import core.basesyntax.service.UserDataInvalidException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final int VALID_USER_AGE = 19;
+    private static final String VALID_USER_LOGIN = "MyLongLogin";
+    private static final String VALID_USER_PASSWORD = "MyLongPassword";
+    private static final int INVALID_USER_AGE = 2;
+    private static final String INVALID_USER_LOGIN = "login";
+    private static final String INVALID_USER_PASSWORD = "pass";
+    private static final long USER_ID = 123L;
+    private static final int DUPLICATE_VALID_USER_AGE = 20;
+    private static final String DUPLICATE_VALID_USER_LOGIN = "LongLogin";
+    private static final String DUPLICATE_VALID_USER_PASSWORD = "LongPassword";
+    private static final long DUPLICATE_USER_ID = 321L;
+    private RegistrationServiceImpl registrationService;
```

Comment:
Consider which way is better to initialize your `RegistrationServiceImpl` class instance: `@BeforeEach` or `@BeforeAll`, what's the difference? Using `@BeforeEach` will create a new instance before each test, which is not necessary if the `RegistrationServiceImpl` does not maintain any state between tests. If it's stateless, `@BeforeAll` could be more efficient.